### PR TITLE
Ensure request and response subscribers complete when channel is unregistered.

### PR DESCRIPTION
**Pull request by chang-chao**
In some cases , such as https://github.com/r2dbc/r2dbc-postgresql/issues/17 , PostgreSQL server does not close the connection gracefully , where `responseProcessor` can't receive the `onComplete` signal. This causes the `Connection.close()` to never end.
This PR fixes this by ensuring connection subscribers to the end in case of channel deregistration.

### CODE REVIEW

File: src/main/java/io/r2dbc/postgresql/client/ReactorNettyClient.java [hunk 1]
```diff
@@ -107,6 +109,16 @@
             sink.next(message);
         });
 
+    private class EnsureSubsribersCompleteChannelHandler extends ChannelDuplexHandler {
```
Comment (nebhale) [inline src/main/java/io/r2dbc/postgresql/client/ReactorNettyClient.java]: Spelling error.

Comment (chang-chao) [reply src/main/java/io/r2dbc/postgresql/client/ReactorNettyClient.java]: fixed

File: src/main/java/io/r2dbc/postgresql/client/ReactorNettyClient.java [hunk 2]
```diff
@@ -111,6 +113,18 @@
             sink.next(message);
         });
 
+    private class EnsureSubscribersCompleteChannelHandler extends ChannelDuplexHandler {
+
+        @Override
+        public void channelUnregistered(ChannelHandlerContext ctx) throws Exception {
+            super.channelUnregistered(ctx);
+            requestProcessor.onComplete();
+            for (MonoSink<Flux<BackendMessage>> responseReceiver = responseReceivers.poll(); responseReceiver != null; ) {
```
Comment (nebhale) [inline src/main/java/io/r2dbc/postgresql/client/ReactorNettyClient.java]: Note that this still needed `responseReceivers.poll()` as the third argument to the for loop to go to the next element in the `Queue`.  I'll fix it in a following commit.

Comment (chang-chao) [reply src/main/java/io/r2dbc/postgresql/client/ReactorNettyClient.java]: Oh，forgot that，thanks！

**Pull request closed without merging**